src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -21,27 +21,103 @@ public function __construct(Client $guzzleClient)
     /**
      * @throws GuzzleException
      */
-    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = []): ResponseInterface
+    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
     {
-        $request = new Request(
-            'GET',
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              [
+                'Authorization' => sprintf('Bearer %s', $bearer),
+                'Accept' => 'application/json',
+              ],
+              json_encode($bodyParams)
+          );
+        } else {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              ['Authorization' => sprintf('Bearer %s', $bearer)]
+          );
+        }
+
+        return $this->guzzleClient->send($request);
```

Comment:
This entire `if` block is copy/pasted in all the methods here, with the HTTP method being the only change that I can see. I recommend moving this into a private method that, along with all the other arguments, takes an string HTTP method. Then, these methods would be much cleaner and look something like this:

```
protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
{
    return $this->sendToApi('GET', $uriEndpoint, $bearer, $queryParamsMap, $bodyParams);
}

protected function deleteApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
{
    return $this->sendToApi('DELETE', $uriEndpoint, $bearer, $queryParamsMap, $bodyParams);
}

// etc.

private function sendToApi(string $httpMethod, string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface 
{
    if (count($bodyParams) > 0) {
        // same code, but use $httpMethod
    } else {
        // same code, but use $httpMethod
    }

    return $this->guzzleClient->send($request);
}
```